modules/nw-ingress-controller-nodeportservice-projects.adoc

```diff
@@ -0,0 +1,138 @@
+// Module included in the following assemblies:
+//
+// * networking/configuring_ingress_cluster_traffic/nw-configuring-ingress-controller-endpoint-publishing-strategy.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="nw-ingress-controller-nodeportservice-projects_{context}"]
+= Adding a single NodePort service to an Ingress Controller
+
+Instead of creating a `NodePort`-type `Service` for each project, you can create a single service and apply it to all of your projects. You can configure this type of environment by setting your Ingress Controller to use the `NodePortService` endpoint publishing strategy.
+
+Consider this configuration for your Ingress Controller when you want to apply a set of routes, through Ingress sharding, to a node even if the node gets displaced. 
+
+Before you `NodePort`-type `Service` for each project, read the following considerations:
+
+* You must create a wildcard DNS record for the Ingress Controller.
+* You must create a route service and specify the hostname of your Ingress Controller in this service.
+* must preserve the non-default port in the curl in order to access, for example: curl <route>:<port> // what does this mean?
+* Ensure that Ingress Controller routes can access your {product-title} cluster through the host IP address assigned to each node.
+
+.Prerequisites
+
+* You installed the {oc-first}.
+* Logged in as a user with `cluster-admin` privileges.
+* You created a wildcard DNS record.
+// https://docs.openshift.com/container-platform/4.16/networking/ingress-controller-dnsmgt.html (does not detail how to create the DNS)
+
+.Procedure
+
+. Create a custom resource (CR) file for the Ingress Controller:
++
+.Example of a CR file that defines information for the Ingress Controller
+[source,yaml]
+----
+apiVersion: v1
+items:
+- apiVersion: operator.openshift.io/v1
+  kind: IngressController
+  metadata:
+    name: custom-ic3 <1>
+    namespace: openshift-ingress-operator
+  spec:
+    replicas: 1
+    domain: my-custom-ic.ipi-cluster.example.com <2>
+    nodePlacement:
+            nodeSelector:
+              matchLabels:
+                ingresscontroller: "custom-ic3" <3>
+    namespaceSelector:
+     matchLabels:
+       ic: custom-ic3 <4>
+    endpointPublishingStrategy:
+      type: NodePortService 
+kind: List
+metadata:
+  resourceVersion: ""
+  selfLink: ""
+# ...
+----
+<1> Specify the `name` with a name for the `IngressController` object.
+<2> Specify the `domain` for the wilcard DNS record.
+<3> Specify the label for the `ingresscontroller` CR.
+<4> Matches routes from any namespaces with this label.
+
+. Add a label to a node by using the `oc label node` command:
```

Comment:
All these commands are needed?

Reply:
Yes, in this case the command would be: 
` oc label node openshift-worker-0 ingresscontroller=custom-ic3`
FYI this needs to match the matchLabels in:
    nodePlacement:
            nodeSelector:
              matchLabels:
                ingresscontroller: "custom-ic3"

Reply:
:heavy_check_mark: